### ISSUE TITLE
Simulate prism observations

### ISSUE THREAD

**Issue by rmjarvis** [GalSim-developers/GalSim#1018]
Rebekah Hounsell wants to use GalSim to simulate prism observations from Roman.  This mostly involves implementing an old (and closed because part of it was done) [issue of 1-d degenerate objects](#533).  In this case, the 1-d function would be the SED in the direction of the prism dispersion.

**Comment by matroxel** [comment]
Working on this with SN PIT. First prototype prism model implemented as photon op in https://github.com/matroxel/roman_imsim/blob/prism/roman_imsim/photonOps.py

Need to update some roman module things to enable prism elements that we had previously never implemented, which will happen in this branch.

**Comment by arunkannawadi** [comment]
I think I'll take this issue over from Troxel. I have computed sky backgrounds corresponding to prism and grism, but noticing that I don't get the current values for the other filters exactly. I know they are accurate to only 10% or so, but I was hoping to get the exact values reproduced by existing filters as a cross check. I'll raise a PR later today so we can see if they seem reasonable.

**Comment by arunkannawadi** [comment]
Troxel tells me that @rmandelb is integrating the code to compute the sky backgrounds on the fly instead of using a lookup table. That's great!

But in the mean time, could we not use the sky backgrounds for `SNPrism` that are already here? I see that it was added in #1017 however, the sky level for prism and grism bands continue to be skipped here: https://github.com/GalSim-developers/GalSim/blob/releases/2.5/galsim/roman/roman_bandpass.py#L138-L139

Can we remove those lines?

**Comment by rmandelb** [comment]
Hi @arunkannawadi - For now, for various practical reasons, I have integrated the code to compute sky backgrounds on the fly (based on Troxel's port of Chris's ETC code, with both of their permission) into a separate repo in which we are developing tools for Euclid-like simulations.  However, the idea is to make that public soon.  If we then want to move that out of the Euclid-like code base and into GalSim, that would be fine.

**Comment by rmjarvis** [comment]
I believe some people have been implementing prism and grism action using photon operators.  Regardless, I think any effort in this direction will happen outside of GalSim proper, so we can close this issue.  (Of course, feel free to reopen if you think there is something useful to be done in GalSim iteself.)

**Comment by arunkannawadi** [comment]
Yes, this is indeed happening outside of GalSim, but I found that some calls to the bandpasses should return the non-imaging Roman bands too to do that consistently. It's fine to close the issue as broadly stated, and I will raise a PR soon.

**Comment by rmjarvis** [comment]
Great.  Thanks Arun!
